code/script.js

```diff
@@ -221,123 +268,244 @@ const CHARACTERS = [
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
-
-  // Invoke a function to redraw the board with the remaining people.
+  else if (group === 'hairColor') {
+    if (keep) { 
+      alert(
+        `Yes, the person has ${questions.options[questions.selectedIndex].value} hair! Keep all persons with ${attribute} hair`
+      )
+    } else {
+      alert(
+        `No, the person doesn't have ${questions.options[questions.selectedIndex].value} hair! Remove all persons with ${attribute} hair`
+      )
+    }
+  } 
+  else if (group === 'eyeColor') {
+    if (keep) { 
+      alert(
+        `Yes, the person has eyecolor ${questions.options[questions.selectedIndex].value}! Keep all persons with eyecolor ${attribute}`
+      )
+    } else {
+      alert(
+        `No, the person doesn't have eyecolor ${questions.options[questions.selectedIndex].value}! Remove all persons with eyecolor ${attribute} `
+      )
+    } 
+  } 
+  else if (group === 'earrings' || group === 'beard') {
+    if (keep) { 
+      alert(
+        `Yes, the person wears ${questions.options[questions.selectedIndex].value}! Keep all persons with ${group}`
+        )
+    } else {
+      alert(
+        `No, the person doesn't wear ${questions.options[questions.selectedIndex].value}! Remove all persons with ${group} `
+      )
+    } 
+  } 
+  generateBoard(); 
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (suspect) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  if (confirm('Are you sure you want to guess?')) {
+    console.log('User want to make a guess!');
+    checkMyGuess(suspect)
+  } else {
+    console.log('The game continues!');
+  }
 }
 
 // If you confirm, this function is invoked
 const checkMyGuess = (suspect) => {
-  // 1. Check if the suspect is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+  console.log(suspect);
+  if (secret.name === suspect) {
+    let showWinner = () => { 
+      document.getElementById('winOrLose').style.display = "block";
+      document.getElementById('showWinOrLose').style.display = "block"
+      winOrLoseText.innerHTML = `Congrats! You won the game!`;
+    } 
+    showWinner()
+    console.log('you won')
+  } else {
+      let showLoser = () => {
+        document.getElementById('winOrLose').style.display = "block";
+        document.getElementById('showWinOrLose').style.display = "block"
+        winOrLoseText.innerHTML = `Sorry, that was a wrong guess! You lost!`;
+      } 
+      showLoser()
+      console.log('you lost')
+    }
+    document.getElementById('board').style.display = "none"; //hides board
```

Comment:
I like the idea of having two functions! And i got a great comment also that stayed with me from one of our classmates. Since the element id for winOrLose is going to be shown either way (even if you win or loose) you dont need to add it multiple times. So what i did was to keep it outside of the if condition. This makes the code a little bit more clean. :)